### PR TITLE
Plugins: add build info to plugin metadata

### PR DESCRIPTION
**Pull request by ryantxu**
While working on a plugin build system, it would be nice to have a standard way to know exactly what version your plugin is.

This adds:
```
export interface PluginSourceInfo {
  time?: number;
  repo?: string;
  branch?: string;
  hash?: string;
}
```

to the info section.  This is all optional, so private repos don't need to expose it if they don't want to.  Or could expose the hash without the repo and branch info

### CODE REVIEW

**Comment by ryantxu** [comment]
This is now generated in the plugin CI.  For example:
https://373-192459187-gh.circle-artifacts.com/0/home/circleci/plugin/ci/report.json

image

but needs this PR to be merged for the info to show up in the API

**Comment by ryantxu** [comment]
should we consider adding this to 6.3 so that it will be exposed to the plugin test framework before 6.4?

I don't think we need to promote this in the changelog since it is really an under-the-hood thing aimed to help testing and validation.

File: pkg/plugins/models.go [hunk 1]
```diff
@@ -112,11 +112,19 @@ type PluginDependencyItem struct {
...
+	Branch string `json:"branch,omitempty"`
+	Hash   string `json:"hash,omitempty"`
+}
+
 type PluginInfo struct {
 	Author      PluginInfoLink      `json:"author"`
 	Description string              `json:"description"`
 	Links       []PluginInfoLink    `json:"links"`
 	Logos       PluginLogos         `json:"logos"`
+	Source      PluginSourceInfo    `json:"source"`
```
Comment (torkelo) [inline pkg/plugins/models.go]: Maybe PluginBuildInfo  is better? Feels more about when and from what is was built

**Review by torkelo — APPROVED** [review]
Good idea, just had a alternative name suggestion

Regarding 6.3 ,yea should be safe to cherry pick

File: pkg/plugins/models.go (hunk 1 above)
Comment (ryantxu) [reply pkg/plugins/models.go]: 👍

File: pkg/plugins/models.go [hunk 2]
```diff
@@ -112,11 +112,19 @@ type PluginDependencyItem struct {
...
+	Branch string `json:"branch,omitempty"`
+	Hash   string `json:"hash,omitempty"`
+}
+
 type PluginInfo struct {
 	Author      PluginInfoLink      `json:"author"`
 	Description string              `json:"description"`
 	Links       []PluginInfoLink    `json:"links"`
 	Logos       PluginLogos         `json:"logos"`
+	Build       PluginBuildInfo     `json:"source"`
```
Comment (torkelo) [inline pkg/plugins/models.go]: You forgot to rename the json property to build

Comment (ryantxu) [reply pkg/plugins/models.go]: yes I did!

**Pull request merged**